Rakefile

```diff
@@ -1,52 +1,123 @@
-#  are permitted provided that the following conditions are met:
-#
-#  (1) Redistributions of source code must retain the above copyright notice,
-#  this list of conditions and the following disclaimer.
-#
-#  (2) Redistributions in binary form must reproduce the above copyright notice,
-#  this list of conditions and the following disclaimer in the documentation and/or
-#  other materials provided with the distribution.
-#
-#  (3) Neither the name of the copyright holder nor the names of any contributors
-#  may be used to endorse or promote products derived from this software without
-#  specific prior written permission from the respective party.
-#
-#  (4) Other than as required in clauses (1) and (2), distributions in any form of
-#  modifications or other derivative works may not use the "BuildingSync" trademark or
-#  any other confusingly similar designation without specific prior written permission
-#  from Alliance for Sustainable Energy, LLC.
-#
-#  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDER(S) AND ANY CONTRIBUTORS "AS IS" AND
-#  ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
-#  WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED.
-#  IN NO EVENT SHALL THE COPYRIGHT HOLDER(S), ANY CONTRIBUTORS, THE UNITED STATES
-#  GOVERNMENT, OR THE UNITED STATES DEPARTMENT OF ENERGY, NOR ANY OF THEIR EMPLOYEES,
-#  BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
-#  DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
-#  LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
-#  THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
-#  NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN
-#  IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
-########################################################################################################################
+require 'bundler'
+Bundler.setup
 
-require 'bundler/gem_tasks'
 require 'rspec/core/rake_task'
 
-RSpec::Core::RakeTask.new(:spec)
+# Always create spec reports
+require 'ci/reporter/rake/rspec'
+
+# Gem tasks
+require 'bundler/gem_tasks'
+
+RSpec::Core::RakeTask.new('spec:unit') do |spec|
+  spec.rspec_opts = %w[--format progress --format CI::Reporter::RSpec]
+  spec.pattern = FileList['spec/**/*_spec.rb']
+end
 
-# Load in the rake tasks from the base openstudio-extension gem
-# Also set measures_dir location for this gem for openstudio-extension gem to use
-require 'openstudio/extension/rake_task'
-require 'buildingsync'
-rake_task = OpenStudio::Extension::RakeTask.new
-rake_task.set_extension_class(BuildingSync::BuildingSync)
+task 'spec:unit' => 'ci:setup:rspec'
+task 'spec:unit' => 'init_config'
+task default: 'spec:unit'
 
 require 'rubocop/rake_task'
-RuboCop::RakeTask.new
+desc 'Run RuboCop on the lib directory'
```

Comment:
@TMaile we should be pulling in rake tasks from the extension gem (as we are here https://github.com/NREL/openstudio-model-articulation-gem/blob/develop/Rakefile#L44) rather than redefining them.  Note that we have restructured all of our extension gems.  The main file, `lib/buildingsync.rb`, should just include other files in the `lib/buildingsync/` directory.  The extension class should be renamed `BuildingSync::Extension` and moved to `lib/buildingsync/extension.rb`.

@nllong FYI, hope we're not giving conflicting input here

Reply:
this is completed as far as I can tell now